linux/src/socket_transport_linux.c

```diff
@@ -191,6 +191,42 @@ SOCKET_TRANSPORT_HANDLE socket_transport_create_server(void)
     return result;
 }
 
+SOCKET_TRANSPORT_HANDLE socket_transport_create_from_socket(SOCKET_HANDLE socket_handle)
+{
+    SOCKET_TRANSPORT* result;
+
+    if (socket_handle == INVALID_SOCKET)
+    {
+        LogError("Invalid socket, unable to create socket_transport_handle.");
+    }
+    else
+    {
+        result = malloc(sizeof(SOCKET_TRANSPORT));
+        if (result == NULL)
+        {
+            LogError("failure allocating SOCKET_TRANSPORT: %zu", sizeof(SOCKET_TRANSPORT));
+        }
+        else
+        {
+            result->sm = sm_create("Socket_transport_win32");
+            if (result->sm == NULL)
+            {
+                LogError("sm_create failed.");
+            }
+            else
```

Comment:
>else [](http://example.com/codeflow?start=12&length=4)

You need to call sm_open_begin here just like you do on line 703 and then you need to call sm_open_end with true.